### PR TITLE
Improving the project template testing tooling 

### PR DESCRIPTION
**Pull request by jmprieur**
Improving the testing tooling to make it easier to test ASP.NET Core project templates

### CODE REVIEW

File: ProjectTemplates/test-templates-from-nuget.bat [hunk 1]
```diff
@@ -2,11 +2,15 @@ echo "Ensure ClientSemVer"
 if "%ClientSemVer%" == "" (
 set ClientSemVer=1.1.0
 )
+
+REM: This is to test Microsoft.Identity.Web templates
 Set TemplateNugetPackageName="Microsoft.Identity.Web.ProjectTemplates"
+Set templatePostFix=2
```
Comment (jmprieur) [inline ProjectTemplates/test-templates-from-nuget.bat:8]: With this script we can now test both our templates and ASP.NET Core templates

File: ProjectTemplates/test-templates-from-nuget.bat [hunk 2]
```diff
@@ -280,7 +284,7 @@ cd ..
 cd ..
 
 echo "Configure the applications"
-..\..\..\..\tools\ConfigureGeneratedApplications\bin\Debug\net5.0\ConfigureGeneratedApplications.exe
+..\..\..\..\tools\ConfigureGeneratedApplications\bin\Debug\netcoreapp3.1\ConfigureGeneratedApplications.exe
```
Comment (jmprieur) [inline ProjectTemplates/test-templates-from-nuget.bat:287]: When testing preview templates the tooling caused issue. Falling back to building it only on .NET Core 3.1 to make things easier

File: tools/ConfigureGeneratedApplications/Model/Replacement.cs [hunk 3]
```diff
@@ -5,6 +5,11 @@ namespace ConfigureGeneratedApplications.Model
 {
     public class Replacement
     {
+        public Replacement()
```
Comment (jmprieur) [inline tools/ConfigureGeneratedApplications/Model/Replacement.cs:8]: In .NET core 3.1, System.Text.Json was less clever, and needed a public parameterless constructor

File: tools/ConfigureGeneratedApplications/Program.cs [hunk 4]
```diff
@@ -96,7 +97,7 @@ private static void ProcessFile(Configuration configuration, string filePath, Fi
                         element = prop.Value;
                     }
 
-                    string replaceFrom = element.ValueKind == JsonValueKind.Number ? element.GetInt32().ToString() : element.ToString();
+                    string replaceFrom = element.ValueKind == JsonValueKind.Number ? element.GetInt32().ToString(CultureInfo.InvariantCulture) : element.ToString();
```
Comment (jmprieur) [inline tools/ConfigureGeneratedApplications/Program.cs:100]: This is enforced in .NET 5

**Review by pmaytak — APPROVED** [review]
👍

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Pull request merged**